agents/ahv/fence_ahv.py

```diff
@@ -0,0 +1,688 @@
+            options(dict): options dictionary
+
+        Returns:
+            None
+        """
+        vmid = None
+        name = None
+        power_state = None
+
+        if "--uuid" in options:
+                vmid = options["--uuid"]
+        else:
+                name = options["--plug"]
+
+        if not vmid and not name:
+                logging.error("Need VM name or VM UUID for power op")
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+        try:
+                power_state = client.get_power_state(vm_name=name, vm_uuid=vmid)
+        except AHVFenceAgentException:
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+        return power_state
+
+def set_power_state(client, options):
+        """
+        Set power state of a VM
+
+        Args:
+            Client(V4NutanixClient instance): V4NutanixClient instance
+            options(dict): options dictionary
+
+        Returns:
+            None
+        """
+        vmid = None
+        name = None
+        action = None
+        timeout = None
+
+        if "--action" not in options:
+                logging.error("No power op action specified")
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+        action = options["--action"].lower()
+
+        if "--power-timeout" in options:
+                timeout = options["--power-timeout"]
+
+        if "--uuid" in options:
+                vmid = options["--uuid"]
+        else:
+                name = options["--plug"]
+
+        if not name and not vmid:
+                logging.error("Need VM name or VM UUID to set power state of a VM")
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+        try:
+                client.set_power_state(vm_name=name, vm_uuid=vmid,
+                                       power_state=action, timeout=timeout)
+        except AHVFenceAgentException as err:
+                logging.error(err)
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+def power_cycle(client, options):
+        """
+        Power cycle a VM
+
+        Args:
+            Client(V4NutanixClient instance): V4NutanixClient instance
+            options(dict): options dictionary
+
+        Returns:
+            None
+        """
+        vmid = None
+        name = None
+        timeout = None
+
+        if "--power-timeout" in options:
+                timeout = options["--power-timeout"]
+
+        if "--uuid" in options:
+                vmid = options["--uuid"]
+        else:
+                name = options["--plug"]
+
+        if not name and not vmid:
+                logging.error("Need VM name or VM UUID to set power cycling a VM")
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+        try:
+                client.power_cycle_vm(vm_name=name, vm_uuid=vmid, timeout=timeout)
+        except AHVFenceAgentException as err:
+                logging.error(err)
+                fencing.fail(fencing.EC_GENERIC_ERROR)
+
+def define_new_opts():
```

Comment:
what is the purpose of this function? is it required?

Reply:
This is for fencing.py. We need to define some options that can be set by pacemaker during configuration.